slither/detectors/functions/codex.py

```diff
@@ -52,29 +53,63 @@ def _detect(self) -> List[Output]:
         openai.api_key = api_key
 
         for contract in self.compilation_unit.contracts:
-            prompt = "Is there a vulnerability in this solidity contracts?\n"
+            if self.slither.codex_contracts != "all" and contract.name not in self.slither.codex_contracts.split(","):
+                continue
+            prompt = "Analyze this Solidity contract and find the vulnerabilities. If you find any vulnerabilities, begin the response with {}".format(VULN_FOUND)
             src_mapping = contract.source_mapping
             content = contract.compilation_unit.core.source_code[src_mapping.filename.absolute]
             start = src_mapping.start
             end = src_mapping.start + src_mapping.length
             prompt += content[start:end]
-            answer = openai.Completion.create(  # type: ignore
-                model="text-davinci-003", prompt=prompt, temperature=0, max_tokens=200
-            )
-
-            if "choices" in answer:
-                if answer["choices"]:
-                    if "text" in answer["choices"][0]:
-                        if "Yes," in answer["choices"][0]["text"]:
-                            info = [
-                                "Codex detected a potential bug in ",
-                                contract,
-                                "\n",
-                                answer["choices"][0]["text"],
-                                "\n",
-                            ]
-
-                            res = self.generate_result(info)
-                            results.append(res)
-
+            logging.info("Querying OpenAI")
+            print("Querying OpenAI")
+            answer = ""
+            res = {}
+            try:
```

Comment:
I added this try/catch because I was hitting the max token limit with a large contract in the prompt.  Another reason to consider running this per-function.